raise_search/api/main.py

```diff
@@ -1,8 +1,63 @@
-from fastapi import FastAPI
+from fastapi import FastAPI, Query
+from fastapi.middleware.cors import CORSMiddleware
 
 app = FastAPI(title="RAISE Search API")
 
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
 
-@app.get("/")
-def read_root():
-    return {"Hello": "World"}
+
+@app.get("/v1/search")
+def search_v1(
+    q: str = Query(..., title="Query string", description="The search query"),
+    version: str = Query(..., title="Version", description="API version"),
+    filter: str = Query(None, title="Filter",
+                        description="Filter by 'student' or 'teacher'"),
+):
+    teacher_only = False
+    if filter and filter == "teacher":
+        teacher_only = True
+
+    search_api_query = {
+        "highlight": {
+            "fields": {
+                "visible_content": {},
+                "lesson_page": {},
+                "activity_name": {},
+            },
+            "pre_tags": ["<strong>"],
+            "post_tags": ["</strong>"],
+        },
+        "query": {},
+        "size": 100
+    }
+
+    if filter is None:
+        search_api_query["query"] = {
+            "simple_query_string": {
+                "query": q,
+                "fields": ["visible_content", "lesson_page", "activity_name"],
+                "default_operator": "AND"
+            }
+        }
+    else:
+        search_api_query["query"] = {
+            "bool": {
+                "filter": {
+                    "term": {"teacher_only": teacher_only}
+                },
+                "must": {
+                    "simple_query_string": {
+                        "query": q,
+                        "fields": ["visible_content", "lesson_page", "activity_name"],
+                        "default_operator": "AND"
+                    }
+                }
+            }
+        }
+
+    return search_api_query
```

Comment:
I realize this may be premature since this PR hasn't been marked as ready for review, but I wanted to confirm there isn't a large disconnect after taking a quick look when I saw it appear in `#k12-stream`: Currently you're just returning the query JSON versus talking to OpenSearch and returning the result. I realize now that perhaps in the issue where I stated:

> The endpoint should convert the request into an Open API query 

it may have been misleading :sweat_smile: . We'll want this endpoint to actually invoke an `opensearch_client` and use it to talk to OpenSearch. We can reference that client via a FastAPI `Depends` for testing + since we're likely going to need to configure that client slightly differently for dev vs prod so it'll be good to have it brought in via a function. We can then define tests which assert on the query JSON the client is invoked with depending on the API arguments.

A couple of other quick thoughts:

* The `version` parameter description is kind of incorrect to me as it's not the API version, but the Content version that the query should be run against
* Is it possible to define `filter` in a way where FastAPI can perform validation / confirm only valid values are provided (and also auto-document valid values)?
* Do we want to define a schema for the response object to help with validation / ensure we're passing only what we expect from the raw OpenSearch response?